Fix compile time warning in dev

Currently when I try to compile I am getting lot of warnings incompatible pointers format issue etc,

<Details>

```c
vibhoothiiaanand@derpBox /V/b/m/C-lings (dev)> gcc compile.c
compile.c:25:21: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'char *'
      [-Wincompatible-pointer-types]
            sprintf(directory, dir_list.array[j]);
                    ^~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:28: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^~~
compile.c:25:32: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
            sprintf(directory, dir_list.array[j]);
                               ^~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
compile.c:25:32: note: treat the string as an argument to avoid this
            sprintf(directory, dir_list.array[j]);
                               ^
                               "%s",
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^
compile.c:30:25: warning: more '%' conversions than data arguments [-Wformat]
                scanf("%c");
                       ~^
compile.c:95:13: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'char *'
      [-Wincompatible-pointer-types]
    sprintf(gcc,"%s%s","gcc ",paths);
            ^~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:28: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^~~
compile.c:95:31: warning: format specifies type 'char *' but the argument has type 'char **' [-Wformat]
    sprintf(gcc,"%s%s","gcc ",paths);
                   ~~         ^~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
compile.c:96:20: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'const char *'
      [-Wincompatible-pointer-types]
    int j = system(gcc);
                   ^~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdlib.h:190:25: note:
      passing argument to parameter here
int      system(const char *) __DARWIN_ALIAS_C(system);
                            ^
compile.c:122:35: warning: incompatible pointer types passing 'char **' to parameter of type 'char *'; dereference with *
      [-Wincompatible-pointer-types]
    sprintf(file_name,"%s",strtok(files,"\n"));
                                  ^~~~~
                                  *
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:90:20: note:
      passing argument to parameter '__str' here
char    *strtok(char *__str, const char *__sep);
                      ^
compile.c:122:13: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'char *'
      [-Wincompatible-pointer-types]
    sprintf(file_name,"%s",strtok(files,"\n"));
            ^~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:28: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^~~
compile.c:123:20: warning: format specifies type 'char *' but the argument has type 'char **' [-Wformat]
    printf("%s \n",files);
            ~~     ^~~~~
compile.c:124:19: warning: incompatible pointer types passing 'char **' to parameter of type 'const char *'; dereference
      with * [-Wincompatible-pointer-types]
   if((fp = fopen(files,"r")) == NULL)
                  ^~~~~
                  *
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:153:37: note:
      passing argument to parameter '__filename' here
FILE    *fopen(const char * __restrict __filename, const char * __restrict __mode) __DARWIN_ALIAS_STARTING(__MAC_...
                                       ^
10 warnings generated.
```

<Details>

--- a/compile.c
+++ b/compile.c
@@ -29,7 +29,8 @@ int main(int argc,char **argv)
         while(j<exe_cnt)
          {
             sprintf(directory,"%s", dir_list.array[j]);
-            compile(directory);
+            printf("inside while %s", directory);
+	    compile(directory);
             if(Complete(directory) != 1)
             {
                 printf("Remove //I AM NOT DONE comment to continue compilation \n");
@@ -42,7 +43,8 @@ int main(int argc,char **argv)
     }
     else if(strcmp("verify",argv[1]) == 0)
     {
-        for(int j=1; j<exe_cnt; j++)
+        printf("\n Inside verify %s \n",directory);
+	for(int j=1; j<exe_cnt; j++)
         {
             sprintf(directory,"%s",dir_list.array[j]);
             compile(directory);
@@ -57,9 +59,10 @@ int main(int argc,char **argv)
 }
  struct lists get_directory(char *path)
 {
-    char cwd[PATH_MAX],command[128],*line =NULL;
+    char cwd[PATH_MAX] = {0}, command[128] = {0}, *line = NULL;
     size_t len = 0;
     struct lists result;
+    printf("Value passed in get_directory: :%s \n",path);
     if (getcwd(cwd, sizeof(cwd)) != NULL)
     {
         printf("problem- \n");
@@ -68,27 +71,26 @@ int main(int argc,char **argv)
     {
         perror("getcwd() error");
     }
-        strcat(command,"ls ");
-        strcat(command,cwd);
-        strcat(command,"/Problems/ > file.txt");
+        sprintf(command,"%s%s%s","ls ",cwd,"/Problems/ > file.txt");
         system(command);
+	printf("\n Total command after strcats: %s \n", command);
         FILE *fp;
         if((fp = fopen("file.txt","r"))== NULL)
         {
             perror("Unable to Open file");
             exit(1);
         }
-        strcat(path,cwd);
-        strcat(path,"/Problems/");
-        int j=0;
+	printf("\n path %s, cwd: %s \n",path,cwd);
+	sprintf(path,"%s%s",cwd,"/Problems/");
+	printf("late path: %s \n", path);
+        int j = 0;
         while(getline(&line,&len,fp)!= -1){
-                strcat(result.array[j],path);
-                strcat(result.array[j],line);
-                printf("%s",result.array[j]);
+                sprintf(result.array[j], "%s%s", path, line);
+		printf("%s",result.array[j]);
             
             j++;       
     }
-        exe_cnt =j;
+        exe_cnt = j;
         return result;
         //sprintf(path,"%s%s%d%s",cwd,"/Problems/",i,".c");
         //printf("%s\n", path);
@@ -96,8 +98,8 @@ int main(int argc,char **argv)
 }
 void compile(char *paths)
 {
-    char h[64];
-    char hint[64];
+    char h[64] = {0};
+    char hint[64] = {0};
     char *gcc = malloc(128 * sizeof(char));
 
     sprintf(gcc,"%s%s","gcc ",paths);
@@ -123,7 +125,7 @@ void compile(char *paths)
 
 int Complete(char *files)
 {
-   char temp[512];
+   char temp[512] = {0};
    char *file_name = malloc( 128 * sizeof(char));
    FILE *fp;
    char *str = "//I AM NOT DONE";

--- a/compile.c
+++ b/compile.c
@@ -4,39 +4,43 @@
 #include <unistd.h>
 #include <limits.h>
 
+void get_directory(char *, int);
+void compile();
 int main()
 {
-  int i, j;
-  char k[64];
-  char x[64];
-  char y[64];
-  char h[64];
-  char hint[64];
-  char gcc[512];
-  char p[64];
-  char cwd[PATH_MAX];
-  if (getcwd(cwd, sizeof(cwd)) != NULL)
-  {
-  }
-  else
-  {
-    perror("getcwd() error");
-    return 1;
-  }
-  for (i = 1; i < 4; i++)
-  {
-    sprintf(x, "%d", i);
-    strcpy(k, "gcc ");
-    strcat(k, cwd);
-    strcpy(p, "/Problems/");
-    strcat(k, p);
-    strcpy(y, ".c");
-    strcat(k, x);
-    strcat(k, y);
-    printf("%s", k);
-    sprintf(gcc, "%s", k);
+    int i;
+    for (i = 1; i < 4; i++)
+    {
+        char *directory[512];
+        get_directory((char *)directory, i);
+        compile(directory);
+    }
+}
+void get_directory(char *path, int i)
+{
+    char cwd[PATH_MAX];
+    if (getcwd(cwd, sizeof(cwd)) != NULL)
+    {
+        printf("problem-");
+    }
+    else
+    {
+        perror("getcwd() error");
+    }
 
-    j = system(gcc);
+        sprintf(path,"%s%s%d%s",cwd,"/Problems/",i,".c");
+        printf("%s\n", path);
+
+}
+void compile(char *paths[512])
+{
+    char h[64];
+    char hint[64];
+    char *gcc[512];
+    
+    sprintf(gcc,"%s%s","gcc ",paths);
+    int j = system(gcc);
+    printf("%d", j);
 
     if (j == 0)
     {

--- a/compile.c
+++ b/compile.c
@@ -7,6 +7,7 @@ struct lists
 {
     char array[50][100];
 };
+
 int exe_cnt =0; // Total number of excerises
 struct lists get_directory(char *);
 void compile();
@@ -18,7 +19,7 @@ int main(int argc,char **argv)
         printf("Enter a valid argument verify/watch \n");
         exit(-1);
     }  
-    char *directory[512];
+    char *directory = malloc(128 * sizeof(char));
     struct lists dir_list;
     dir_list = get_directory((char *)directory); 
  
@@ -27,7 +28,7 @@ int main(int argc,char **argv)
         int j=0;
         while(j<exe_cnt)
          {
-            sprintf(directory, dir_list.array[j]);
+            sprintf(directory,"%s", dir_list.array[j]);
             compile(directory);
             if(Complete(directory) != 1)
             {
@@ -43,7 +44,7 @@ int main(int argc,char **argv)
     {
         for(int j=1; j<exe_cnt; j++)
         {
-            sprintf(directory, (char *)dir_list.array[j]);
+            sprintf(directory,"%s",dir_list.array[j]);
             compile(directory);
         }
     }
@@ -93,11 +94,11 @@ int main(int argc,char **argv)
         //printf("%s\n", path);
 
 }
-void compile(char *paths[512])
+void compile(char *paths)
 {
     char h[64];
     char hint[64];
-    char *gcc[512];
+    char *gcc = malloc(128 * sizeof(char));
 
     sprintf(gcc,"%s%s","gcc ",paths);
     int j = system(gcc);
@@ -120,10 +121,10 @@ void compile(char *paths[512])
     }
 }
 
-int Complete(char *files[512])
+int Complete(char *files)
 {
    char temp[512];
-   char *file_name[512];
+   char *file_name = malloc( 128 * sizeof(char));
    FILE *fp;
    char *str = "//I AM NOT DONE";
     sprintf(file_name,"%s",strtok(files,"\n"));

--- a/compile.c
+++ b/compile.c
@@ -3,39 +3,94 @@
 #include <string.h>
 #include <unistd.h>
 #include <limits.h>
-
-void get_directory(char *, int);
+struct lists
+{
+    char array[50][100];
+};
+int exe_cnt =0; // Total number of excerises
+struct lists get_directory(char *);
 void compile();
 int Complete();
-int main()
+int main(int argc,char **argv)
 {
-    int i;
-    for (i = 1; i < 4; i++)
+    if(argc == 1)
     {
-        char *directory[512];
-        get_directory((char *)directory, i);
-        compile(directory);
-        if(Complete(directory) != 1)
-        {
-            printf("Remove //I AM NOT DONE comment to continue compilation \n");
-            break;
+        printf("Enter a valid argument verify/watch \n");
+        exit(-1);
+    }  
+    char *directory[512];
+    struct lists dir_list;
+    dir_list = get_directory((char *)directory); 
+ 
+    if(strcmp("watch",argv[1]) == 0)
+    {
+        int j=0;
+        while(j<exe_cnt)
+         {
+            sprintf(directory, dir_list.array[j]);
+            compile(directory);
+            if(Complete(directory) != 1)
+            {
+                printf("Remove //I AM NOT DONE comment to continue compilation \n");
+                char *u;
+                scanf("%c",u);
+            }
+            else j++;
+            
         }
     }
+    else if(strcmp("verify",argv[1]) == 0)
+    {
+        for(int j=1; j<exe_cnt; j++)
+        {
+            sprintf(directory, (char *)dir_list.array[j]);
+            compile(directory);
+        }
+    }
+    else
+    {
+        printf("Commnad not found \n");
+    }
+    
+    
 }
-void get_directory(char *path, int i)
+ struct lists get_directory(char *path)
 {
-    char cwd[PATH_MAX];
+    char cwd[PATH_MAX],command[128],*line =NULL;
+    size_t len = 0;
+    struct lists result;
     if (getcwd(cwd, sizeof(cwd)) != NULL)
     {
-        printf("problem-");
+        printf("problem- \n");
     }
     else
     {
         perror("getcwd() error");
     }
-
-        sprintf(path,"%s%s%d%s",cwd,"/Problems/",i,".c");
-        printf("%s\n", path);
+        strcat(command,"ls ");
+        strcat(command,cwd);
+        strcat(command,"/Problems/ > file.txt");
+        system(command);
+        FILE *fp;
+        if((fp = fopen("file.txt","r"))== NULL)
+        {
+            perror("Unable to Open file");
+            exit(1);
+        }
+        strcat(path,cwd);
+        strcat(path,"/Problems/");
+        int j=0;
+        while(getline(&line,&len,fp)!= -1){
+                strcat(result.array[j],path);
+                strcat(result.array[j],line);
+                printf("%s",result.array[j]);
+            
+            j++;       
+    }
+        exe_cnt =j;
+        return result;
+        //sprintf(path,"%s%s%d%s",cwd,"/Problems/",i,".c");
+        //printf("%s\n", path);
 
 }
 void compile(char *paths[512])
@@ -71,10 +126,9 @@ int Complete(char *files[512])
    char *file_name[512];
    FILE *fp;
    char *str = "//I AM NOT DONE";
-   
-   sprintf(file_name,"%s",files);
-
-   if((fp = fopen(file_name,"r")) == NULL)
+    sprintf(file_name,"%s",strtok(files,"\n"));
+    printf("%s \n",files);
+   if((fp = fopen(files,"r")) == NULL)
     {    
        perror("File Opening Error");
        return(-1);

--- a/compile.c
+++ b/compile.c
@@ -6,6 +6,7 @@
 
 void get_directory(char *, int);
 void compile();
+int Complete();
 int main()
 {
     int i;
@@ -14,6 +15,11 @@ int main()
         char *directory[512];
         get_directory((char *)directory, i);
         compile(directory);
+        if(Complete(directory) != 1)
+        {
+            printf("Remove //I AM NOT DONE comment to continue compilation \n");
+            break;
+        }
     }
 }
 void get_directory(char *path, int i)
@@ -37,10 +43,10 @@ void compile(char *paths[512])
     char h[64];
     char hint[64];
     char *gcc[512];
-    
+
     sprintf(gcc,"%s%s","gcc ",paths);
     int j = system(gcc);
-    printf("%d", j);
+    //printf("%d", j);
 
     if (j == 0)
     {
@@ -57,7 +63,33 @@ void compile(char *paths[512])
         printf("try checking print statement\n");
       }
     }
-  }
-
-  return 0;
 }
+
+int Complete(char *files[512])
+{
+   char temp[512];
+   char *file_name[512];
+   FILE *fp;
+   char *str = "//I AM NOT DONE";
+   
+   sprintf(file_name,"%s",files);
+
+   if((fp = fopen(file_name,"r")) == NULL)
+    {    
+       perror("File Opening Error");
+       return(-1);
+    }
+   while(fgets(temp,512,fp) != NULL){
+       if(strstr(temp,str) != NULL)
+       {
+               return(0);
+       }
+   }
+   if(fp)
+    {
+       fclose(fp);
+    }
+   return(1);
+  
+}
+

